sdk/client-core/core/src/main/java/io/clientcore/core/http/models/RequestOptions.java

```diff
@@ -205,6 +211,10 @@ public RequestOptions setHeader(HttpHeaderName header, String value) {
      * @return The updated {@link RequestOptions} object.
      */
     public RequestOptions addQueryParam(String parameterName, String value) {
+        if (locked) {
+            throw LOGGER.logThrowableAsError(new IllegalStateException("RequestOptions.NONE is immutable."));
```

Comment:
Adding the operation name in the exception message could help troubleshooting. `RequestOptions.NONE is immutable. Cannot add query params`. We should update the message in other places too.